# Food Data Central Exploration Notebook

## Set Up

#### Import pandas and limit the data display for large queries to 50 rows.

#### Then load all the necessary files.

In [2]:
import pandas as pd

pd.set_option('display.max_row', 50)

In [3]:
# Load all of our files
food_data = pd.read_csv("srlegacy/food.csv")
food_nutrient = pd.read_csv("srlegacy/food_nutrient.csv")
food_attribute = pd.read_csv("srlegacy/food_attribute.csv")
food_category = pd.read_csv("srlegacy/food_category.csv")
nutrient = pd.read_csv("srlegacy/nutrient.csv")


## Time to explore the data

#### Let's see how the data in these files is structured.

In [4]:
food_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7793 entries, 0 to 7792
Data columns (total 5 columns):
fdc_id              7793 non-null int64
data_type           7793 non-null object
description         7793 non-null object
food_category_id    7793 non-null int64
publication_date    7793 non-null object
dtypes: int64(2), object(3)
memory usage: 304.5+ KB


In [5]:
food_nutrient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 644125 entries, 0 to 644124
Data columns (total 11 columns):
id                   644125 non-null int64
fdc_id               644125 non-null int64
nutrient_id          644125 non-null int64
amount               644125 non-null float64
data_points          644125 non-null int64
derivation_id        462382 non-null float64
min                  96891 non-null float64
max                  96893 non-null float64
median               0 non-null float64
footnote             264 non-null object
min_year_acquired    0 non-null float64
dtypes: float64(6), int64(4), object(1)
memory usage: 54.1+ MB


In [6]:
food_attribute.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1074 entries, 0 to 1073
Data columns (total 6 columns):
id                        1074 non-null int64
fdc_id                    1074 non-null int64
seq_num                   1074 non-null int64
food_attribute_type_id    1074 non-null int64
name                      0 non-null float64
value                     1070 non-null object
dtypes: float64(1), int64(4), object(1)
memory usage: 50.5+ KB


In [7]:
food_category.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
id             28 non-null int64
code           28 non-null int64
description    28 non-null object
dtypes: int64(2), object(1)
memory usage: 800.0+ bytes


In [26]:
food_category

id  code                          description
0    1   100               Dairy and Egg Products
1    2   200                     Spices and Herbs
2    3   300                           Baby Foods
3    4   400                        Fats and Oils
4    5   500                     Poultry Products
5    6   600           Soups, Sauces, and Gravies
6    7   700          Sausages and Luncheon Meats
7    8   800                    Breakfast Cereals
8    9   900              Fruits and Fruit Juices
9   10  1000                        Pork Products
10  11  1100    Vegetables and Vegetable Products
11  12  1200                Nut and Seed Products
12  13  1300                        Beef Products
13  14  1400                            Beverages
14  15  1500       Finfish and Shellfish Products
15  16  1600          Legumes and Legume Products
16  17  1700        Lamb, Veal, and Game Products
17  18  1800                       Baked Products
18  19  1900                               Sweets
19  20  2000              Cereal Grains and Pasta
20  21  2100                           Fast Foods
21  22  2200      Meals, Entrees, and Side Dishes
22  23  2500                               Snacks
23  24  3500  American Indian/Alaska Native Foods
24  25  3600                     Restaurant Foods
25  26  4500       Branded Food Products Database
26  27  2600            Quality Control Materials
27  28  1410                  Alcoholic Beverages

In [8]:
nutrient.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 227 entries, 0 to 226
Data columns (total 5 columns):
id              227 non-null int64
name            227 non-null object
unit_name       227 non-null object
nutrient_nbr    227 non-null float64
rank            226 non-null float64
dtypes: float64(2), int64(1), object(2)
memory usage: 9.0+ KB


#### Let's explore the data from the food table

In [11]:
food_data.head(10)

fdc_id       data_type                                        description  \
0  167512  sr_legacy_food  Pillsbury Golden Layer Buttermilk Biscuits, Ar...   
1  167513  sr_legacy_food  Pillsbury, Cinnamon Rolls with Icing, refriger...   
2  167514  sr_legacy_food  Kraft Foods, Shake N Bake Original Recipe, Coa...   
3  167515  sr_legacy_food     George Weston Bakeries, Thomas English Muffins   
4  167516  sr_legacy_food         Waffles, buttermilk, frozen, ready-to-heat   
5  167517  sr_legacy_food  Waffle, buttermilk, frozen, ready-to-heat, toa...   
6  167518  sr_legacy_food  Waffle, buttermilk, frozen, ready-to-heat, mic...   
7  167519  sr_legacy_food    Waffle, plain, frozen, ready-to-heat, microwave   
8  167520  sr_legacy_food  Pie Crust, Cookie-type, Graham Cracker, Ready ...   
9  167521  sr_legacy_food     Pie Crust, Cookie-type, Chocolate, Ready Crust   

   food_category_id publication_date  
0                18       2019-04-01  
1                18       2019-04-01  
2                18       2019-04-01  
3                18       2019-04-01  
4                18       2019-04-01  
5                18       2019-04-01  
6                18       2019-04-01  
7                18       2019-04-01  
8                18       2019-04-01  
9                18       2019-04-01

#### Suppose we wanted to find the nutritional content of waffles.

#### First, let's search for 'Waffles' in the description of our food table.

In [14]:
waffles_data = food_data[food_data.description.str.contains('Waffles')]
waffles_data

fdc_id       data_type  \
4     167516  sr_legacy_food   
12    167524  sr_legacy_food   
499   168011  sr_legacy_food   
6573  174085  sr_legacy_food   
6593  174105  sr_legacy_food   
7526  175038  sr_legacy_food   
7527  175039  sr_legacy_food   
7536  175048  sr_legacy_food   

                                            description  food_category_id  \
4            Waffles, buttermilk, frozen, ready-to-heat                18   
12       Waffles, chocolate chip, frozen, ready-to-heat                18   
499        Van's, Gluten Free, Totally Original Waffles                18   
6573  Waffles, whole wheat, lowfat, frozen, ready-to...                18   
6593        Waffles, gluten-free, frozen, ready-to-heat                18   
7526              Waffles, plain, frozen, ready-to-heat                18   
7527               Waffles, plain, prepared from recipe                18   
7536    Waffles, plain, frozen, ready -to-heat, toasted                18   

     publication_date  
4          2019-04-01  
12         2019-04-01  
499        2019-04-01  
6573       2019-04-01  
6593       2019-04-01  
7526       2019-04-01  
7527       2019-04-01  
7536       2019-04-01

#### The food table contains several varieties of waffles. We would need to have some way to determine which one we are interested in.

#### Let's choose the first one (Waffles, buttermilk, froen, ready-to-heat)

#### Let's find out what nutrients are contained in those waffles. To do this, we need the fdc_id for that particular waffle (167516), and we need to link to the food_nutrient table.

In [15]:
food_nutrient[food_nutrient.fdc_id.eq(167516)]

id  fdc_id  nutrient_id    amount  data_points  derivation_id  min  \
62   1283736  167516         1258     1.898            0           49.0  NaN   
63   1283737  167516         1104  1340.000            0           49.0  NaN   
64   1283738  167516         1106   401.000            0           49.0  NaN   
65   1283739  167516         1293     1.445            0           49.0  NaN   
66   1283740  167516         1190    96.000            0           49.0  NaN   
..       ...     ...          ...       ...          ...            ...  ...   
168  1283842  167516         1005    41.050            0           49.0  NaN   
169  1283843  167516         1212     0.281            0           39.0  NaN   
170  1283844  167516         1213     0.489            0           39.0  NaN   
171  1283845  167516         1214     0.296            0           39.0  NaN   
172  1283846  167516         1216     0.154            0           39.0  NaN   

     max  median footnote  min_year_acquired  
62   NaN     NaN      NaN                NaN  
63   NaN     NaN      NaN                NaN  
64   NaN     NaN      NaN                NaN  
65   NaN     NaN      NaN                NaN  
66   NaN     NaN      NaN                NaN  
..   ...     ...      ...                ...  
168  NaN     NaN      NaN                NaN  
169  NaN     NaN      NaN                NaN  
170  NaN     NaN      NaN                NaN  
171  NaN     NaN      NaN                NaN  
172  NaN     NaN      NaN                NaN  

[111 rows x 11 columns]

#### That particular waffle contains 111 nutrients.

#### If we want to know what those nutrients are, we need to link to the nutrient table.

In [18]:
nutrient[nutrient.id.isin( food_nutrient[food_nutrient.fdc_id.eq(167516)].nutrient_id   )]

id                          name unit_name  nutrient_nbr     rank
1    1003                       Protein         G         203.0    600.0
2    1004             Total lipid (fat)         G         204.0    800.0
3    1005   Carbohydrate, by difference         G         205.0   1110.0
4    1007                           Ash         G         207.0   1000.0
5    1008                        Energy      KCAL         208.0    300.0
..    ...                           ...       ...           ...      ...
180  1321                18:3 n-6 c,c,c         G         685.0  14100.0
181  1323                          17:1         G         687.0  12000.0
182  1325                          20:3         G         689.0  14400.0
187  1333                          15:1         G         697.0  11600.0
201  2000  Sugars, total including NLEA         G         269.0   1510.0

[111 rows x 5 columns]

#### Let's run the above query again with only the first 5 nutrients from our food_nutrient table.

In [27]:
nutrient[nutrient.id.isin( [1258, 1104, 1106, 1293, 1190]  )]

id                                name unit_name  nutrient_nbr     rank
49   1104                       Vitamin A, IU        IU         318.0   7500.0
51   1106                      Vitamin A, RAE        UG         320.0   7420.0
98   1190                         Folate, DFE        UG         435.0   7200.0
134  1258        Fatty acids, total saturated         G         606.0   9700.0
163  1293  Fatty acids, total polyunsaturated         G         646.0  12900.0

#### Based on the above analysis, we can make the below statements about frozen, ready-to-heat, buttermilk waffles:

* They contain 111 nutrients.
* They contain 1340 IU of Vitamin A, IU
* They contain 401 UG of Vitamin A, RAE
* They contain 96 UG of Folate, DFE
* They contain 1.898 grams of Fatty acids, total saturated
* They contain 1.445 grams of Fatty acids, total polyunsaturated

*(Note: The values given are per 100 grams of the food consumed)*